specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-10-01-preview/fluxconfiguration.json

```diff
@@ -0,0 +1,786 @@
+            },
+            "repositoryPublicKey": {
+              "description": "Public Key associated with this fluxConfiguration (either generated within the cluster or provided by the user).",
+              "type": "string",
+              "x-nullable": true,
+              "readOnly": true
+            },
+            "lastSourceSyncedCommit": {
+              "description": "Branch and SHA of the last source commit synced with the cluster.",
+              "type": "string",
+              "x-nullable": true,
+              "readOnly": true
+            },
+            "lastSourceSyncedAt": {
+              "description": "Datetime the fluxConfiguration last synced its source on the cluster.",
+              "type": "string",
+              "x-nullable": true,
+              "readOnly": true
+            },
+            "complianceState": {
+              "description": "Combined status of the Flux Kubernetes resources created by the fluxConfiguration or created by the managed objects.",
+              "type": "string",
+              "readOnly": true,
+              "$ref": "#/definitions/FluxComplianceStateDefinition"
+            },
+            "provisioningState": {
+              "description": "Status of the creation of the fluxConfiguration.",
+              "type": "string",
+              "readOnly": true,
+              "$ref": "../../common/2021-10-01-preview/definitions.json#/definitions/ProvisioningStateDefinition"
+            },
+            "errorMessage": {
+              "description": "Error message returned to the user in the case of provisioning failure.",
+              "type": "string",
+              "x-nullable": true,
+              "readOnly": true
+            }
+          }
+        },
+        "systemData": {
+          "description": "Top level metadata https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-contracts.md#system-metadata-for-all-azure-resources",
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "FluxConfigurationPatch": {
+      "description": "The Flux Configuration Patch Request object.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "type": "object",
+          "x-ms-client-flatten": true,
+          "description": "Updatable properties of an Flux Configuration Patch Request",
+          "properties": {
+            "sourceKind": {
+              "description": "Source Kind to pull the configuration data from.",
+              "$ref": "#/definitions/SourceKindDefinition"
+            },
+            "suspend": {
+              "description": "Whether this configuration should suspend its reconciliation of its kustomizations and sources.",
+              "type": "boolean",
```

Comment:
>boolean [](http://example.com/codeflow?start=23&length=7)

String enums are preferred over boolean. Generally enums are more descriptive, flexible and future-proof than booleans and easier for customers to discover, understand and use. In this case consider reconciliation: [ Running, Suspended ] or similar.

Reply:
If we used enums, the values would be something like `Enabled` and `Disabled` which would effectively translate to boolean values. Is the recommendation still to do something like `Enabled` and `Disabled` as opposed to booleans?